integrations/statuspage/client.py

```diff
@@ -0,0 +1,115 @@
+from enum import StrEnum
+from typing import Any, AsyncGenerator, Optional
+from loguru import logger
+from port_ocean.utils import http_async_client
+
+PAGE_SIZE = 50
+
+
+class ResourceKey(StrEnum):
+    COMPONENT_GROUPS = "component-groups"
+    COMPONENTS = "components"
+    INCIDENTS = "incidents"
+
+
+class StatusPageClient:
+    def __init__(
+        self, statuspage_api_key: str, statuspage_ids: Optional[list[str]] = None
+    ) -> None:
+        self.statuspage_url = "https://api.statuspage.io/v1/pages"
```

Comment:
Move the host as an optional configuration to the integration, with a default value of `https://api.statuspage.io/`